lib/smb2/client.rb

```diff
@@ -1,7 +1,21 @@
 require 'smb2/packet'
+require 'smb2/tree'
 require 'net/ntlm'
 require 'net/ntlm/client'
 
+# A client for holding the state of an SMB2 session.
+#
+# ```ruby
```

Comment:
Usage example in YARD should go in [`@example` tags](http://www.rubydoc.info/gems/yard/file/docs/Tags.md#example):

```ruby
# A client for holding the state of an SMB2 session.
#
# @example Authenticating to a SMB2 server with username, password, and domain
#     sock = TCPSocket.new("192.168.100.140", 445)
#     c = Smb2::Client.new(
#       socket: sock,
#       username:"administrator",
#       password:"P@ssword1",
#       domain:"asdfasdf"
#     )
#     c.negotiate
#     c.authenticate
```